src/writer.rs

```diff
@@ -170,6 +168,22 @@ impl<W: Write> Writer<W> {
         }
         Ok(())
     }
+
+    /// Write event nested in another level:
+    /// ```xml
+    /// <parent_name>{event}</parent_name>
+    /// ```
+    pub fn write_nested_event<'a, N: AsRef<[u8]>, E: AsRef<Event<'a>>>(
+        &mut self,
+        parent_name: N,
+        event: E,
+    ) -> Result<()> {
+        let name = parent_name.as_ref();
+        self.write_event(Event::Start(BytesStart::borrowed(name, name.len())))?;
+        self.write_event(event)?;
+        self.write_event(Event::End(BytesEnd::borrowed(name)))?;
+        Ok(())
+    }
```

Comment:
Would it be possible to allow setting attributes without making it too difficult to use? e.g.

```xml
  <checksum type="sha256">fc83446cfb15634f0200a29a0138e22a011ea4a064d531c0c53402b56a46ccab</checksum>
```

An API that emulates the [Entry API](https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html) for hash maps might be nice (but probably a lot more work).  Maybe something like this:

```rust
writer.create_node("checksum")
      .push_attribute(("type", "sha256")
      .nested_event(Event::Text(...));
```

Reply:
That's a good suggestion let me try

Reply:
Are you currently working on it?  If not, I could give it a try

Reply:
Sorry I got something more urgent to do at work. Feel free to give it a try I'll review.
Thanks!

Reply:
@tafia I gave it a shot, let me know what you think.  It's definitely not perfect but I'm posting it so I can go to sleep :)

https://github.com/tafia/quick-xml/pull/278